base/MAPL_EASEGridFactory.F90

```diff
@@ -0,0 +1,1866 @@
+              staggerloc=ESMF_STAGGERLOC_CORNER, &
+              farrayPtr=corners, _RC)
+         do j = 1, size(centers,2)
+            centers(:,j) = this%lon_centers(i_1:i_n)
+         end do
+         do j = 1, size(corners,2)
+            corners(:,j) = this%lon_corners(ic_1:ic_n)
+         end do
+
+         ! Now latitudes
+         call ESMF_GridGetCoord(grid, coordDim=2, localDE=0, &
+              staggerloc=ESMF_STAGGERLOC_CENTER, &
+              farrayPtr=centers, _RC)
+         call ESMF_GridGetCoord(grid, coordDim=2, localDE=0, &
+              staggerloc=ESMF_STAGGERLOC_CORNER, &
+              farrayPtr=corners, _RC)
+
+         do i = 1, size(centers,1)
+            centers(i,:) = this%lat_centers(j_1:j_n)
+         end do
+         do i = 1, size(corners,1)
+            corners(i,:) = this%lat_corners(jc_1:jc_n)
+         end do
+      end if
+
+      _RETURN(_SUCCESS)
+
+   end subroutine add_horz_coordinates
+
+
+   subroutine initialize_from_file_metadata(this, file_metadata, unusable, force_file_coordinates, rc)
+      use MAPL_KeywordEnforcerMod
+      use MAPL_BaseMod, only: MAPL_DecomposeDim
+
+      class (EASEGridFactory), intent(inout)  :: this
+      type (FileMetadata), target, intent(in) :: file_metadata
+      class (KeywordEnforcer), optional, intent(in) :: unusable
+      logical, optional, intent(in) :: force_file_coordinates
+      integer, optional, intent(out) :: rc
+
+      integer :: status
+
+      class (CoordinateVariable), pointer :: v
+      class (*), pointer :: ptr(:)
+
+      character(:), allocatable :: lon_name
+      character(:), allocatable :: lat_name
+      character(:), allocatable :: lev_name
+      character(:), allocatable :: grid_name
+      integer :: i
+      logical :: hasLon, hasLat, hasLongitude, hasLatitude, hasLev,hasLevel,regLat,regLon
+      real(kind=REAL64) :: del12,delij
+
+      integer :: i_min, i_max
+      real(kind=REAL64) :: d_lat, d_lat_temp, extrap_lat
+      logical :: is_valid, use_file_coords, compute_lons, compute_lats, has_bnds
+
+      _UNUSED_DUMMY(unusable)
+
+      if (present(force_file_coordinates)) then
+         use_file_coords = force_file_Coordinates
+      else
+         use_file_coords = .false.
+      end if
+
+      ! Cannot assume that lats and lons are evenly spaced
+      this%is_regular = .false.
+
+      associate (im => this%im_world, jm => this%jm_world, lm => this%lm)
+         lon_name = 'lon'
+         hasLon = file_metadata%has_dimension(lon_name)
+         if (hasLon) then
+            im = file_metadata%get_dimension(lon_name, _RC)
+         else
+            lon_name = 'longitude'
+            hasLongitude = file_metadata%has_dimension(lon_name)
+            if (hasLongitude) then
+               im = file_metadata%get_dimension(lon_name, _RC)
+            else
+               _FAIL('no longitude coordinate')
+            end if
+         end if
+
+         grid_name = ease_grid_name(im)
```

Comment:
I'm not sure why we need the EASE grid name here, and how it could be obtained without going through the number of cols.  Perhaps this is unavoidable, but I still like to think that we can obtain the name without going through the number of columns.  Same for the other "initialize_from...()" subroutine below.

Reply:
At some point, if only cols is provided, we can construct the factory and thus EASE grid